### DISCUSSION TITLE
ConnectionRefusedError

### DISCUSSION THREAD

**Discussion by varaprasadredy** [planetscale/discussion#493] [Q&A]
Hello,

I encountered a "ConnectionRefusedError" when attempting to establish a connection from the Nodejs application.

Here I'm adding my codebase and please help me to resolve the issue.

File: db.connection.js 
//Import Modules or Paths
import Sequelize from "sequelize";
import dotenv from "dotenv";
dotenv.config();
import mysql from "mysql2";

const connection = mysql.createConnection(process.env.DATABASE_URL);
//sequelize connection setup for mariadb
const sequelize = new Sequelize({
  dialect: "mysql",
  dialectModule: mysql,
  define: {
    timestamps: false,
  },
  connection: connection,
});

// Sequelize Sync
sequelize
  .sync({ force: false })
  .then(() => {
    console.log(`Database & tables created!`);
  })
  .catch((error) => {
    console.log("Error is................", error);
  });
//Export sequelize connection
export default sequelize;

File: user.mode.js
//Import Modules or Paths
import Sequelize from "sequelize";
import sequelize from "../utility/db.connection.js";

//Create a Schema
const UserSchema = sequelize.define("User", {
  id: {
    type: Sequelize.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  name: {
    type: Sequelize.STRING,
    allowNull: false,
  },
  email: {
    type: Sequelize.STRING,
    allowNull: false,
    required: true,
    unqiue: true,
  },
  password: {
    type: Sequelize.STRING,
    required: true,
    allowNull: false,
  },
});
//Export user schema
export default UserSchema;

**Comment by orware** [comment]
Hello @varaprasadredy,

This may have something to do with your SSL/TLS configuration for the database connection and this may require some additional parameters to be passed into your `DATABASE_URL` value.

Which operating system are you using currently?

It may be that you have to provide the correct path to your operating system's Root CA Certificate file, or you may also download the Let's Encrypt one from this URL:
https://letsencrypt.org/certs/isrgrootx1.pem

And then modify your `mysql.createConnection` line a bit to provide that file for the connection:
```
var connection = mysql.createConnection({
    host: PLANETSCALE_HOSTNAME,
    user: PLANETSCALE_USERNAME,
    password: PLANETSCALE_PASSWORD,
    ssl: {
        ca: fs.readFileSync(__dirname + '/isrgrootx1.pem')
    }
});
```

Within the `ssl` object you may also need to include `rejectUnauthorized: true` but you can try things with just the `ca` value included to see if that's helpful.

Hopefully this is the information you need to get connected successfully!
